scripts/cluster/setup_master_node.go

```diff
@@ -199,18 +215,42 @@ func InstallKnativeServingComponent(stockContainerd string) error {
 		if !utils.CheckErrorWithMsg(err, "Failed to install Knative Serving component!\n") {
 			return err
 		}
-		_, err = utils.ExecShellCmd("kubectl apply -f https://github.com/knative/serving/releases/download/knative-v%s/serving-core.yaml", configs.Knative.KnativeVersion)
-		if !utils.CheckErrorWithTagAndMsg(err, "Failed to install Knative Serving component!\n") {
-			return err
+		
+		if _, fileStat := os.Stat(path.Join(configs.VHive.VHiveRepoPath, path.Join("configs/knative_yamls", "container-serving-core.yaml"))); fileStat != nil {
+			_, err = utils.ExecShellCmd("kubectl apply -f https://github.com/knative/serving/releases/download/knative-v%s/serving-core.yaml", configs.Knative.KnativeVersion)
+			if !utils.CheckErrorWithTagAndMsg(err, "Failed to install Knative Serving component!\n") {
+				return err
+			}
+		} else {
+			servingCorePath, err := utils.GetVHiveFilePath(path.Join("configs/knative_yamls", "container-serving-core.yaml"))
+			if err != nil {
+				return err
+			}
+			_, err = utils.ExecShellCmd("cat %s | envsubst | kubectl apply -f -", servingCorePath)
+			if !utils.CheckErrorWithTagAndMsg(err, "Failed to install Knative Serving component!\n") {
+				return err
+			}
 		}
 	} else {
 		_, err := utils.ExecShellCmd("kubectl apply -f %s/serving-crds.yaml", configs.Knative.NotStockOnlyKnativeServingYamlUrlPrefix)
```

Comment:
Same, rename this file, since it is made for firecracker.

Reply:
Same here, rename it to follow our convention of naming yamls based on function (firecracker here). And make it using local file instead of downloading it from github.